### ISSUE TITLE
[BUG] parinfer-rust throws E565: not allowed to change text or change window

### ISSUE THREAD

**Issue by jakesarjeant** [nix-community/nixvim#1881]
| Field        | Description |
|--------------|-------------|
| Plugin       | `parinfer-rust`  | 
| Nixpkgs      | `24.05` and `unstable` | 
| Home Manager | N/A | 

- [x] I have read the [FAQ](https://nix-community.github.io/nixvim/user-guide/faq.html) and my bug is not listed there.

## Description

I would like to use parinfer-rust in my NixVim config. I enabled the plugin, but upon attempting to type a parenthesis or bracket, parinfer causes the error `E565: Not allowed to change text or change window`. Since I use stable nixpkgs, I tried setting `plugins.parinfer-rust.package` to the unstable version of the package in case this is a bug in parinfer-rust that has since been mitigated, to no avail. I used parinfer-rust in my previous (non-NixVim) config and never had such an issue.

## Minimal, Reproducible Example (MRE)

```nix
programs.nixvim = {
  plugins.parinfer-rust.enable = true;
}
```

**Comment by jakesarjeant** [comment]
By the way: Although I'd rather use parinfer-rust, my current workaround is to just use [nvim-parinfer](https://github.com/gpanders/nvim-parinfer):

```nix
{ pkgs, ... }:
{
  extraPlugins = [(pkgs.vimUtils.buildVimPlugin {
    name = "nvim-parinfer";
    src = pkgs.fetchFromGitHub {
      owner = "gpanders";
      repo = "nvim-parinfer";
      rev = "5ca0928";
      hash = "sha256-diwLtmch8LzaX7FIwBNy78n3iY7VnqMC1n0ep8k5kWE=";
    };
  })];
}
```

**Comment by MattSturgeon** [comment]
This sounds like an upstream bug. Unless you think something nixvim is doing is contributing to the issue?

Has the issue been reported upstream?

**Comment by jakesarjeant** [comment]
I'd be surprised if this is an upstream issue as I've been unable to reproduce it without installing through nixvim – then again, the issue happens even if I install the plugin as an extra, so it still might be. I'm not home right now, but I'll look into it when I get back

**Comment by GaetanLepage** [comment]
Are you sure this bug is not caused by the neovim version being too old (`0.9.5` in nixos-24.05) ?

**Comment by jakesarjeant** [comment]
@GaetanLepage Oops, perhaps I misinterpreted the template. My system is on 24.05, but my NixVim flake uses nixpkgs/master. I'm also pretty sure it's not an issue with that version mismatch since I have the same issue on my mac where I have everything on unstable...

**Comment by MattSturgeon** [comment]
> my NixVim flake uses nixpkgs/master

Unrelated to the main topic, but that sounds like a bad idea; you should probably use the `nixos-unstable` channel or at least something like `nixos-unstable-small` or `nixpkgs-unstable`.

> `plugins.parinfer-rust.enable = true;`

If you can reproduce just with this, then our tests _should_ catch this issue :confused:

**Comment by jakesarjeant** [comment]
> > my NixVim flake uses nixpkgs/master
> 
> Unrelated to the main topic, but that sounds like a bad idea; you should probably use the `nixos-unstable` channel or at least something like `nixos-unstable-small` or `nixpkgs-unstable`.
> 

Fair enough, I wasn't really paying much attention to which channel I was using when I wrote the config

> > `plugins.parinfer-rust.enable = true;`
> 
> If you can reproduce just with this, then our tests _should_ catch this issue :confused:

Okay, then it sounds like it's probably some kind of misconfiguration on my end. I don't really have time to mess with it at the moment but I'll take a look later this afternoon.

**Comment by MattSturgeon** [comment]
> I don't really have time to mess with it at the moment but I'll take a look later this afternoon.

Please don't ever feel under pressure to respond quickly! Take your time, of course!

**Comment by jakesarjeant** [comment]
> Please don't ever feel under pressure to respond quickly! Take your time, of course!

It's really no problem, I'm just out right now so I only have my phone to work with. I just want to do my best to cooperate with your investigation, especially since it seems reasonably likely that I'm the one at fault here, and I don't want to waste more maintainer resources than I need to.
